File: offer05-替换空格/offer05.cpp
#include <iostream>

using namespace std;

char a[1001];

string replaceSpace(string s)
{
	int count = 0;
	int length = s.size();
	for (int i = 0; i < length; i++)
		if (s[i] == ' ')
			count++;
	int len = length + count * 2;
	s.resize(len);
	while (length--)
	{
		if (s[length] == ' ')
			s[--len] = '0', s[--len] = '2', s[--len] = '%';
		else
			s[--len] = s[length];
	}
	return s;
}

int main()
{
	ios::sync_with_stdio(false);
	string s = "We are happy.";
	cout << replaceSpace(s) << endl;
	return 0;
}

File: offer05-替换空格/offer05.go
package main

import "fmt"

func main() {
	s := "We are happy."
	fmt.Println(replaceSpace(s))
}

func replaceSpace(s string) string {
	var res = make([]byte, 0)
	for i := range s {
		if s[i] == ' ' {
			res = append(res, '%', '2', '0')
		} else {
			res = append(res, s[i])
		}
	}
	return string(res)
}
